Add product API generator script and sample JSON

Zelamene opened the pull request:
Added products_api.py to generate fake product data using Faker Included sample_10_PRODUCTS.json for quick tests
Added requirements.txt with dependencies
productsApi.json is gitignored due to large file size

Zelamene committed:
Add product API generator script and sample JSON

Added products_api.py to generate fake product data using Faker
Included sample_10_PRODUCTS.json for quick tests
Added requirements.txt with dependencies
productsApi.json is gitignored due to large file size

logi4ace-coder commented:
Yes, I am still looking for datasets for the images.
I should have everything done by 14:00 today.

CheriseHeyl commented:
Thank you, sounds perfect

Zelamene committed:
Update products_api.py

I fixed how product brands were being generated(the product dataset doesn't provide them) . Previously, almost all products ended up with the same brand, which made it hard to test brand filters properly.

I also added an array of "acceptable" countries for clothing products. These countries were chosen based on how commonly they appear in real clothing labels and their popularity in fashion manufacturing.

Lastly, I added product images using entries from Koogle's dataset.
